roofit/histfactory/src/HistoToWorkspaceFactoryFast.cxx

```diff
@@ -1907,8 +1907,8 @@ namespace HistFactory{
 
     // New Asimov Generation: Use the code in the Asymptotic calculator 
     // Need to get the ModelConfig...
-    RooDataSet* asimov_dataset = (RooDataSet*) AsymptoticCalculator::GenerateAsimovData(*model, observables);
-    proto->import(*asimov_dataset, Rename("asimovData"));
+    unique_ptr<RooAbsData> asimov_dataset(AsymptoticCalculator::GenerateAsimovData(*model, observables));
+    proto->import(dynamic_cast<RooDataSet&>(*asimov_dataset), Rename("asimovData"));
```

Comment:
Does it make sense to use a dyn cast when we do not check its result? Why not simply dereferencing?

Reply:
1. The cast is necessary because import only takes the derived class ...
2. The cast does get checked because it's a reference cast. It will throw a bad cast exception. It will never fail, though, unless someone changes the behaviour of the AsymptoticCalculator. In that case, one would need a way of diagnosing the problem.

I wonder if assertion, crash or exception are the best option?

Reply:
If there is a general strategy of throwing in RooFit, I would say exception. Otherwise we might go for an assert.

Reply:
Maybe @lmoneta wants to comment?